lib/net.js

```diff
@@ -1294,6 +1449,62 @@ function afterConnect(status, handle, req, readable, writable) {
   }
 }
 
+function afterConnectMultiple(context, status, handle, req, readable, writable) {
+  context.connecting--;
+
+  // Some error occurred, add to the list of exceptions
+  if (status !== 0) {
+    let details;
+    if (req.localAddress && req.localPort) {
+      details = req.localAddress + ':' + req.localPort;
+    }
+    const ex = exceptionWithHostPort(status,
+                                     'connect',
+                                     req.address,
+                                     req.port,
+                                     details);
+    if (details) {
+      ex.localAddress = req.localAddress;
+      ex.localPort = req.localPort;
+    }
+
+    ArrayPrototypePush(context.errors, ex);
+
+    if (context.connecting === 0) {
+      this.destroy(aggregateErrors(context.errors));
+    }
+
+    return;
+  }
+
+  // One of the connection has completed and correctly dispatched, ignore this one
+  if (context.completed) {
+    debug('connect/happy eyeballs: ignoring successful connection to %s:%s', req.address, req.port);
+    handle.close();
```

Comment:
This looks like it fully establishes more than one connection. But shouldn't the final ACK only be sent to the first connection that responds with SYN-ACK? (Originally asked in https://github.com/nodejs/node/pull/44731#discussion_r982871113.)

Reply:
Can we control that from JS?

Reply:
I don't think so - See https://github.com/nodejs/node/pull/44731#discussion_r983408158